assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -1400,4 +1426,34 @@ export default Component.extend({
       document.documentElement.scrollTo(0, 0);
     }
   },
+
+  _setupVisibleMessagesObserver() {
+    const options = {
+      root: document.querySelector(".chat-messages-container"),
+      rootMargin: "0px",
+      threshold: 1.0,
+    };
+
+    const callback = (entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          entry.target.classList.add("visible");
```

Comment:
It's better to use a data attribute here instead of a class because classes a) are more commonly used for styling (and in this case we don't want to add styling) and b) avoids name clashes in our stylesheets. The "visible" class is indeed used in a bunch of places in core and plugins, so avoiding it here would be best.